### ISSUE TITLE
Release 2.6

### ISSUE THREAD

**Issue by camilasan** [nextcloud/desktop#1369]
### 2.6
- [X] RC1: 26.08.19
- [X] Final Release: after two weeks time (max) of latest RC release. (27.09.2019)

### 📝 TO DO
- [x] [Milestone: 2.6](https://github.com/nextcloud/desktop/milestone/8)
- [x] 👥 Pull requests:
     - [x] [label:backport-request](https://github.com/nextcloud/desktop/issues?q=is%3Aopen+is%3Apr+label%3Abackport-request) 
     - [x] [milestone:"2.6"](https://github.com/nextcloud/desktop/pulls?q=is%3Aopen+is%3Apr+milestone%3A%222.6+Login+flow+v2%22)  
- [x] When needed: backport master merged PR's using the backport bot by commenting:
```
/backport to stable-2.6.0
```
- [x] Create tag from branch ```stable-2.6``` created previously
```
$ git checkout stable-2.6.0
$ git tag -s v2.6.0-rc1 -m "Release..."
$ git checkout -b stable-2.6.0-rc1
$ git push origin --tags
```
- [x] 🏭 Login on the build servers to change the scripts with the current tag and version suffix to build the release:
    - [x] via Parallels for the Mac OS
    - [x] via RDP for Windows
    - [x] via ssh for Linux      
- [x] Move new pkg/exe/AppImage it to [download.nextcloud.com/desktop/releases](https://download.nextcloud.com/desktop/releases/)
- [x] Get changelog using [github_helper scripts](https://github.com/nextcloud/github_helper/tree/master/changelog)
- [X] Create pre-release/release at [nextcloud/desktop/releases](https://github.com/nextcloud/desktop/releases) with the new changelog.
- [X] 🔔 Notify Marketing to share the release in the proper channels.

### ⚠️ TO DO before major releases
- [x] 🍴 Create stable branch (e.g. ```stable-2.6.0```) after tagging RC1 for major release (e.g. ```v2.6.0-rc1```)
- [ ] Update ```CHANGELOG.md``` with the new changelog.
- [x] Bump version in ```master``` in [VERSION.cmake](https://github.com/nextcloud/desktop/blob/master/VERSION.cmake) (e.g. ```2.7.0```)

**Comment by ivaradi** [comment]
If I understand correctly, a separate branch will be used for 2.5.4 and the release will be tagged on this branch.

However, it seems to me, that automatic builds are not run for these branches, only for merges to the master branch and pull requests. It means that the release will not be built for Ubuntu and will not be uploaded to the stable repository.

What could be done to still have a stable Ubuntu build? Could Drone be configured to build for other branches or specifically for such release branches? (I cannot see any option, but I may not have all permissions.) Or could we merge this branch back to master, and tag the release there? Or shall I do a build manually after the release? Or is there a way to start a Drone build manually?

**Comment by camilasan** [comment]
To be honest @ivaradi this is our first try to have a release checklist and follow a proper plan now that we have @misch7 on board. We can change it and make sure Ubuntu has a stable build. Thanks for the feedback, I will get back to you.

**Comment by tobiasKaminsky** [comment]
> * Backport master merged PR's when needed using the backport bot by commenting:

Maybe I misunderstand it, but I would do the backport as soon as a PR is merged and not too late. Otherwise master and stable branch has diverged too much and it is even harder to do the backport.

**Comment by tobiasKaminsky** [comment]
> What could be done to still have a stable Ubuntu build? Could Drone be configured to build for other branches or specifically for such release branches?

https://github.com/nextcloud/desktop/blob/master/.drone.yml search for branch: master.
This means that each pipeline/test is run only on master. 
One you split it up to e.g. stable-2.5.4 you would have to change this.



> Or could we merge this branch back to master, and tag the release there?

Tag will be done in stable-2.5.4 branch, but you can check out the tag and it does not matter on which branch it is. (at least on Android we are doing it this way)

**Comment by tobiasKaminsky** [comment]
@camilasan @misch7 if you want to follow "our" Android way, then the patch-level (.4) in 2.5.4 is only for urgent bug fixing, e.g. data loss.
Otherwise we are always increasing minor level (2.5.0 -> 2.6.0), following semantic version.
Then you would have stable-2.6 branch, where you tag 2.6.0 from. And if needed 2.6.1,… and in parallel you have master preparing for 2.7.

**Comment by camilasan** [comment]
I see my mistake. Thanks @tobiasKaminsky <3

**Comment by camilasan** [comment]
I have updated the checklist: we create stable-2.5 and tag 2.5.x from there. But it doesn't solve @ivaradi's problem, correct?

**Comment by camilasan** [comment]
> > * Backport master merged PR's when needed using the backport bot by commenting:
> 
> Maybe I misunderstand it, but I would do the backport as soon as a PR is merged and not too late. Otherwise master and stable branch has diverged too much and it is even harder to do the backport.

If I branch 2.5 and tag from there...in the meantime, whatever is merged in master and needed in 2.5 can be backported.

**Comment by tobiasKaminsky** [comment]
> If I branch 2.5 and tag from there...in the meantime, whatever is merged in master and needed in 2.5 can be backported.

Yes, and I encourage you to do it as soon as possible, as then chances of conflicts are less likely.

**Comment by tobiasKaminsky** [comment]
> I have updated the checklist: we create stable-2.5 and tag 2.5.x from there. Please, @tobiasKaminsky check it. 

This is correct.

> But it doesn't solve @ivaradi's problem, correct?

See my answer here: https://github.com/nextcloud/desktop/issues/1369#issuecomment-520303307

**Comment by tobiasKaminsky** [comment]
In overview you say "create stable 2.6". I would do this at the same day when creating 2.6.0 RC1, so in 2-3 months.

**Comment by ivaradi** [comment]
@tobiasKaminsky, do I understand correctly, that the branch in ```.drone.yml``` should simply be changed to, e.g., ```stable-2.5``` on that branch for at least the Debian build? And perhaps also enable ```tag``` for the event so that tagging would trigger a build? (The build script looks for a tag of the right format on the commit to decide if a build should go to the stable or the beta repository.)

**Comment by tobiasKaminsky** [comment]
@ivaradi I do not know your debian build system, but drone is "only" for our continuous integration system. 
In my opinion it should be enough to monitor https://github.com/nextcloud/desktop/tags and check them out and build them.

For building you do not need to change anything:
- git clone https://github.com/nextcloud/desktop/tags
- git fetch --tags
- git checkout v2.5.3
- then follow build instructions

**Comment by ivaradi** [comment]
@tobiasKaminsky the idea is that for each pull request or merge to master an automatic build is started in Drone, and it includes a build of the Debian/Ubuntu source packages. If the build is for a PR, the packages are not uploaded to Launchpad, but if it is for a merge, they are uploaded. In that case the last commit is checked, and if it is also tagged with a tag of format "vA.B.C", the destination of the uploaded is the stable repository, otherwise the beta one.

So while a stable build could be built by hand, why not build it with the automatic build system?

**Comment by tobiasKaminsky** [comment]
Ah, nice. I think it is a bit overhead, can't you just check for new tags?
You do maintain then your own drone.yml?
As you want to run it any any commit, maybe you can just remove "branch: master". Then drone will run on your side every time.

**Comment by FrostKnight** [comment]
Hey, just wanted to let you all know, that qt problem I have been experiencing after 2.3.2 and onward, where the appimage version of this, doesn't work because it goes by the version of my computer instead of the one in the appimage... you seem to have fixed it in the latest stable build. I don't know what you did, but many times thanks!

Just to be clear what I mean:  https://github.com/nextcloud/desktop/issues/950#issuecomment-449778492

Sorry if this is off topic, but I wanted to make sure you guys knew you fixed an issue so you can continue to have it working, etc...  Again I really appreciate it. :)

**Comment by tflidd** [comment]
>  Update `CHANGELOG.md` with the new changelog.

The changelog is still missing. It looks small and unimportant, it can however help to clear many question in advance when stuff is not working for users, e.g. https://help.nextcloud.com/t/high-memory-load-from-client-with-high-access-requests-apache-access-log/61120

**Comment by tflidd** [comment]
sorry, I didn't want to reopen...
